server/src/main/java/org/elasticsearch/search/retriever/rankdoc/RankDocsQueryBuilder.java

```diff
@@ -46,6 +55,10 @@ RankDoc[] rankDocs() {
     @Override
     protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeArray(StreamOutput::writeNamedWriteable, rankDocs);
+        out.writeOptionalArray(StreamOutput::writeNamedWriteable, queryBuilders);
```

Comment:
Shouldn't this be protected via transport version as well?

Reply:
yeap, missed that :)